common/testing/taskpoller/taskpoller.go

```diff
@@ -262,6 +389,51 @@ func (p *workflowTaskPoller) pollTask(
 	return resp, err
 }
 
+func (p *workflowTaskPoller) HandleQueries(
+	tv *testvars.TestVars,
+	handler func(task *workflowservice.PollWorkflowTaskQueueResponse) (*workflowservice.RespondQueryTaskCompletedRequest, error),
+	opts ...optionFunc,
+) (*workflowservice.RespondQueryTaskCompletedResponse, error) {
+	p.t.Helper()
+	options := newOptions(tv, opts)
+	ctx, cancel := newContext(options)
+	defer cancel()
+	return p.pollAndHandleQueries(ctx, options, handler)
+}
+
+func (p *workflowTaskPoller) pollAndHandleQueries(
+	ctx context.Context,
+	opts *options,
+	handler func(task *workflowservice.PollWorkflowTaskQueueResponse) (*workflowservice.RespondQueryTaskCompletedRequest, error),
+) (*workflowservice.RespondQueryTaskCompletedResponse, error) {
+	p.t.Helper()
+	task, err := p.pollTask(ctx, opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to poll workflow task: %w", err)
+	}
+	return p.handleQueries(ctx, opts, task, handler)
+}
+
+func (p *workflowTaskPoller) handleQueries(
+	ctx context.Context,
+	opts *options,
+	task *workflowservice.PollWorkflowTaskQueueResponse,
+	handler func(task *workflowservice.PollWorkflowTaskQueueResponse) (*workflowservice.RespondQueryTaskCompletedRequest, error),
+) (*workflowservice.RespondQueryTaskCompletedResponse, error) {
+	p.t.Helper()
+	reply, err := handler(task)
+	if err != nil {
+		return nil, p.respondTaskFailed(ctx, opts, task.TaskToken, err)
```

Comment:
So from reading the [API docs](https://github.com/temporalio/api/blob/e8fc1a5c490987714081f2aa0d308601cbf1304b/temporal/api/workflowservice/v1/request_response.proto#L914), it seems to me that the correct response here would be to return a `RespondQueryTaskCompletedRequest` with `ErrorMessage` and `Failure` fields set appropriately.